deeprankcore/trainer.py

```diff
@@ -75,80 +75,110 @@ def __init__( # pylint: disable=too-many-arguments, too-many-branches
             if self.target is None:
                 raise ValueError("No target set. Make sure the pretrained model explicitly defines the target to train against.")
 
-            self._check_dataset_equivalence()
+            self.pretrained_model_path = pretrained_model
+
+            self._load_params()
             self._load_pretrained_model()
 
+    def _init_output_exporters(self, output_exporters: Optional[List[OutputExporter]]):
+
+        if output_exporters is not None:
+            self._output_exporters = OutputExporterCollection(*output_exporters)
+        else:
+            self._output_exporters = OutputExporterCollection(HDF5OutputExporter('./output'))
+
+    def _init_datasets(self,  # pylint: disable=too-many-arguments
+                       train_dataset: Union[GraphDataset, GridDataset],
+                       validation_dataset: Optional[Union[GraphDataset, GridDataset]],
+                       test_dataset: Optional[Union[GraphDataset, GridDataset]],
+                       validation_size: Optional[Union[int, float]],
+                       test_size: Optional[Union[int, float]]):
+
+        self._check_dataset_equivalence(train_dataset, validation_dataset, test_dataset)
+
+        self.dataset_train = train_dataset
+        self.dataset_test = test_dataset
+        self.dataset_val = validation_dataset
+        self.val_size = validation_size
+        self.test_size = test_size
+
+        # Divide datasets where necessary.
+        if test_size is not None:
+            if test_dataset is None:
+                self.dataset_train, self.dataset_test = _divide_dataset(train_dataset, test_size)
+            else:
+                _log.warning("Test dataset was provided to Trainer; test_size parameter is ignored.")
+
+        if validation_size is not None:
+            if validation_dataset is None:
+                self.dataset_train, self.dataset_val = _divide_dataset(train_dataset, validation_size)
+            else:
+                _log.warning("Validation dataset was provided to Trainer; val_size parameter is ignored.")
+
+        # Copy settings from the dataset that we will use.
+        if self.dataset_train is not None:
+            self._init_from_dataset(self.dataset_train)
+        else:
+            self._init_from_dataset(self.dataset_test)
+
+    def _init_from_dataset(self, dataset: Union[GraphDataset, GridDataset]):
+
+        if isinstance(dataset, GraphDataset):
+            self.clustering_method = dataset.clustering_method
+            self.node_features = dataset.node_features
+            self.edge_features = dataset.edge_features
+            self.features = None
+
+        elif isinstance(dataset, GridDataset):
+            self.clustering_method = None
+            self.node_features = None
+            self.edge_features = None
+            self.features = dataset.features
+        else:
+            TypeError(type(dataset))
```

Comment:
```suggestion
            raise TypeError(type(dataset))
```

Reply:
Changed this in commit 9ad784c964ef77876f71e39599c150dcace6bb22.

Reply:
Fixed this in e2b2e6f74e8c13db88326c1c5e17dc89fd77e169.